### PR TITLE
test(ui): add Playwright e2e browser tests for the UI

### PR DESCRIPTION
**Pull request by Joibel**
### Motivation

The UI has ~270 source files but only Jest unit tests covering a fraction of them, and no browser-level end-to-end testing at all — CI only checks that the webpack dev server starts. This adds a Playwright e2e suite so real user journeys are exercised in a browser against the full stack. It also extracts the e2e CI setup that `e2e-tests` and the new `e2e-ui` job would otherwise duplicate into composite actions, so those steps can't drift.

The tests added here are just basic to prove the system works, and see if we're happy with the infratstructure.

I feel like we could lose the old CI/UI test with this.

### Modifications

**Playwright e2e suite (`ui/e2e/`)**

- Config, `global-setup` (client-token cookie auth from the `argo-server` service-account-token secret — the same source as the Go e2e suite — plus first-time/new-version modal suppression), REST-API fixtures that seed workflows and wait for a terminal phase before asserting (with per-test cleanup), page objects, and specs.
- First journeys: login (token auth + logout) and the workflow list (render, name filter, row navigation to details).
- `make test-ui-e2e` target; Jest is configured to ignore `e2e/`; `data-testid`s added to the workflow name filter and the row status cell for stable selectors.

**CI**

- New `e2e-ui` job: brings the stack up in-cluster with Tilt (`--mode=ci`) and runs Playwright against the production UI bundle the argo-server serves at `:2746`. Wired into the existing `e2e-tests-composite-result` required check.
- Extracted the shared cluster setup (disk cleanup, k3d/Tilt install, `k3d-up`, `/etc/hosts`) and the shared failure-debug log dumps from `e2e-tests` and `e2e-ui` into two local composite actions (`.github/actions/e2e-cluster-up`, `.github/actions/e2e-failure-debug`).

### Verification

- Suite runs green locally against `make start` (Tilt); see `ui/e2e/README.md`.
- `yarn --cwd ui test` (Jest, 24 suites / 115 tests) and `yarn --cwd ui lint` pass; Jest does not pick up the Playwright specs.
- The `e2e-ui` job exercises the suite on a k3d + Tilt cluster in CI. The composite-action behaviour can only be fully confirmed on a live CI run, so please watch the first run on this PR.

### Documentation

- `docs/running-locally.md` gains a "Running UI E2E tests locally" section.
- `ui/e2e/README.md` documents the auth/seeding model, the selector policy, and NixOS specifics (running the bundled browser via `nix-ld`).

### AI

Developed with Claude Code (Anthropic). Code, tests, documentation, and commit messages were produced with substantial AI assistance and reviewed by me before submission, in line with the Argo project's Generative AI policy.

## Summary by CodeRabbit

* **New Features**
  * Added browser-based UI end-to-end tests (authentication + workflow list) using Playwright, including richer test reporting and a UI E2E CI job.
  * Introduced standardized composite actions for provisioning an in-cluster environment and collecting E2E failure diagnostics.
  * Added a `test-ui-e2e` Make target to run the UI test suite.

* **Bug Fixes**
  * Updated CI triggers and gating so UI E2E runs when relevant UI changes are detected.
  * Improved E2E failure diagnostics by focusing workflow pod descriptions.

* **Documentation**
  * Expanded local UI E2E setup and troubleshooting guides.

### CODE REVIEW

**Review by terrytangyuan — APPROVED** [review]
This is great to have 👍

**Comment by Joibel** [comment]
@MasonM I'd appreciate you taking a look at this

File: docs/running-locally.md [hunk 1]
```diff
@@ -196,6 +196,35 @@ Tests often fail: that's good. To diagnose failure:
...
+
+```bash
+make start
+```
+
+Then run the tests in another terminal:
+
+```bash
+yarn --cwd ui install
+yarn --cwd ui playwright install chromium   # first run only
```
Comment (MasonM) [inline docs/running-locally.md]: The `--with-deps` is needed in the devcontainer, otherwise you'll get:
```

      - [pid=31750][err] /home/vscode/.cache/ms-playwright/chromium_headless_shell-1228/chrome-headless-shell-linux64/chrome-headless-shell: error while loading shared libraries: libatk-1.0.so.0: cannot open shared object file: No such file or directory
```

```suggestion
yarn --cwd ui playwright install --with-deps chromium   # first run only
```

File: ui/e2e/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+```bash
+make start
+```
+
+Then, in another terminal:
+
+```bash
+yarn --cwd ui install
+yarn --cwd ui playwright install chromium   # first run only
```
Comment (MasonM) [inline ui/e2e/README.md]: Same as above:
```suggestion
yarn --cwd ui playwright install --with-deps chromium   # first run only
```

File: ui/e2e/global-setup.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                origin: url.origin,
+                localStorage: [
+                    {name: 'modal/ftu', value: '"dismissed"'},
+                    {name: 'modal/version', value: JSON.stringify(version)}
+                ]
+            }
+        ]
+    };
+    mkdirSync(AUTH_DIR, {recursive: true});
+    writeFileSync(STORAGE_STATE, JSON.stringify(state, null, 2));
```
Comment (MasonM) [inline ui/e2e/global-setup.ts:71]: This file doesn't seem to be used

File: docs/running-locally.md [hunk 4]
```diff
@@ -196,6 +196,35 @@ Tests often fail: that's good. To diagnose failure:
...
 
+### Running UI E2E tests locally
+
+The UI has browser-level end-to-end tests written with [Playwright](https://playwright.dev), separate from the
+Jest unit tests (`yarn --cwd ui test`). They drive a real browser against a running stack.
+
+Start the dev stack in one terminal — Tilt serves the UI on <http://localhost:8080>:
+
+```bash
+make start
```
Comment (MasonM) [inline docs/running-locally.md]: The default auth mode for `make start` is `hybrid`, but `.github/workflows/ci-build.yaml` uses `--auth-mode=client`, so to match the CI behavior this should be: 
```suggestion
make start AUTH_MODE=client 
```

File: ui/e2e/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+Browser-level end-to-end tests for the Argo Workflows UI. Unlike the Jest unit
+tests (`yarn test`), these drive a real browser against a live stack.
+
+## Running locally
+
+Start the dev stack in one terminal (Tilt serves the UI on http://localhost:8080):
+
+```bash
+make start
```
Comment (MasonM) [inline ui/e2e/README.md]: Same as above:
```suggestion
make start AUTH_MODE=client 
```

**Review by MasonM — APPROVED** [review]
Thanks for doing this! Just some minor comments.

Also, `ui/src/login/login.test.tsx` fails if run locally with `make start AUTH_MODE=client` because there's an uncaught exception, which should be fixed in https://github.com/argoproj/argo-workflows/pull/16491. The reason it's not failing in CI is because the uncaught exception gets ignored in CI, whereas in local development Webpack Dev Server shows a "Uncaught runtime errors" overlay, which prevents Playwright from clicking elements behind the overlay.

**Comment by Joibel** [comment]
> Thanks for doing this! Just some minor comments.
> 
> Also, `ui/src/login/login.test.tsx` fails if run locally with `make start AUTH_MODE=client` because there's an uncaught exception, which should be fixed in #16491. The reason it's not failing in CI is because the uncaught exception gets ignored in CI, whereas in local development Webpack Dev Server shows a "Uncaught runtime errors" overlay, which prevents Playwright from clicking elements behind the overlay.

I think we shouldn't ignore during playwright, that feels wrong. I'll try to switch it.

File: ui/e2e/global-setup.ts (hunk 3 above)
Comment (Joibel) [reply ui/e2e/global-setup.ts:71]: I have added a comment - this is read by playwright, not us.

**Comment by Joibel** [comment]
I have made this fail without #16491, we no longer hide errors under playwright. https://github.com/argoproj/argo-workflows/pull/16398/commits/5abc44271bcc31363dd4dda31ce06614fc18ee66
